tests/test_testclient.py

```diff
@@ -254,19 +254,6 @@ async def asgi(receive, send):
         assert data == {"message": "test"}
 
 
-def test_client(test_client_factory):
-    async def app(scope, receive, send):
-        client = scope.get("client")
-        assert client is not None
-        host, port = client
-        response = JSONResponse({"host": host, "port": port})
-        await response(scope, receive, send)
-
-    client = test_client_factory(app)
-    response = client.get("/")
-    assert response.json() == {"host": "testclient", "port": 50000}
```

Comment:
I didn't see this test before... Can we have a background of why it exists?

Reply:
It was added here: https://github.com/encode/starlette/pull/1462
I think it was not intentional, because for the test client it’s not per spec.